### DISCUSSION TITLE
Can't pull in dependencies for v2.4.2

### DISCUSSION THREAD

**Discussion by jollygreenegiant** [supabase-community/supabase-kt#606] [Q&A]
I'm having trouble pulling in the latest dependencies for v2.4.2.  I'm pulling them in as shown below, and getting the following error:

```
1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':shared:transformNativeMainCInteropDependenciesMetadataForIde'.
> Could not resolve all files for configuration ':shared:iosArm64CompilationDependenciesMetadata'.
   > Could not find storage-kt-iosarm64-2.4.2.klib (io.github.jan-tennert.supabase:storage-kt-iosarm64:2.4.2).
     Searched in the following locations:
         https://plugins.gradle.org/m2/io/github/jan-tennert/supabase/storage-kt-iosarm64/2.4.2/storage-kt-iosarm64-2.4.2.klib
         https://plugins.gradle.org/m2/io/github/jan-tennert/supabase/storage-kt-iosarm64/2.4.2/storage-kt-iosarm64-2.4.2.jar
   > Could not find functions-kt-iosarm64-2.4.2-metadata.jar (io.github.jan-tennert.supabase:functions-kt-iosarm64:2.4.2).
     Searched in the following locations:
         https://plugins.gradle.org/m2/io/github/jan-tennert/supabase/functions-kt-iosarm64/2.4.2/functions-kt-iosarm64-2.4.2-metadata.jar
   > Could not find supabase-kt-iosarm64-2.4.2.klib (io.github.jan-tennert.supabase:supabase-kt-iosarm64:2.4.2).
     Searched in the following locations:
         https://plugins.gradle.org/m2/io/github/jan-tennert/supabase/supabase-kt-iosarm64/2.4.2/supabase-kt-iosarm64-2.4.2.klib
         https://plugins.gradle.org/m2/io/github/jan-tennert/supabase/supabase-kt-iosarm64/2.4.2/supabase-kt-iosarm64-2.4.2.jar
```

This seems to only be having an issue with the iOS dependencies, because that's the only thing it warns about, and if I move the import from commonMain to androidMan (just to see), everything works fine.  It seems like it isn't searching maven for the dependencies on the iOS side for some reason, but if I do 2.4.1 instead, things work fine.  I feel like I'm going crazy, is there something I'm missing?

shared build.gradle:
```
sourceSets {
        val commonMain by getting {
            implementation(project.dependencies.platform("io.github.jan-tennert.supabase:bom:2.4.2"))
            implementation(libs.supabase.postgrest)
            implementation(libs.supabase.functions)
            implementation(libs.supabase.realtime)
            implementation(libs.supabase.storage)
    }
}
```

libs.toml:
```
supabase-functions = { module = "io.github.jan-tennert.supabase:functions-kt" }
supabase-postgrest = { module = "io.github.jan-tennert.supabase:postgrest-kt" }
supabase-realtime = { module = "io.github.jan-tennert.supabase:realtime-kt" }
supabase-storage = { module = "io.github.jan-tennert.supabase:storage-kt" }
```

**Comment by jan-tennert** [comment]
The dependency is definitely there: 
https://central.sonatype.com/artifact/io.github.jan-tennert.supabase/supabase-kt-iosarm64
From the error it seems like its only searching for the dependency in the gradle plugins portal, I assume you have maven central as a repository?

**Reply by jollygreenegiant** [reply]
I noticed that as well, and did check Maven Central and saw this artifact was there, and I am using `mavenCentral()` in the gradle files.  I'm trying to bump my version from 2.1.5 to 2.4.2, were there any breaking changes in that jump that would potentially cause issues with pulling in the dependencies?  I looked through the docs for each release and nothing jumped out at me, but I may have missed something.

2.1.5 imports everything perfectly, changing nothing but the version number to 2.4.2 causes all kinds of issues finding the dependencies for that version.

**Reply by jan-tennert** [reply]
And 2.4.1 works?

**Reply by jollygreenegiant** [reply]
Yes, v2.4.1 syncs with Gradle successfully, but then isn't able to recognize the `storage` or `functions` packages
<img width="469" alt="Screenshot 2024-05-27 at 3 00 20 PM" src="https://github.com/supabase-community/supabase-kt/assets/25214257/1105dbe0-f930-4935-b1c6-0ca3d4cf810b">

**Reply by jan-tennert** [reply]
What is your Kotlin and Gradle version? Can you try to investigate the last version which has no problem?

**Reply by jollygreenegiant** [reply]
Gradle: 8.5
AGP: 8.1.4
Kotlin: 1.9.22

Even just bumping from Supabase-kt 2.1.5 -> 2.1.6 causes the issue here, even after invalidating gradle caches and restarting.  2.1.5 works totally fine, but bumping to 2.1.6 gets the Gradle sync error where it seems to not be searching Maven Central.

**Reply by jollygreenegiant** [reply]
Updated to Gradle 8.6 and AGP 8.4.1, still no luck

**Reply by jollygreenegiant** [reply]
Okay I have seem to have gotten it working.

I removed `gradlePluginPortal()` from `settings.gradle` and added 
```
allProjects {
    repositories {
        mavenCentral()
        gradlePluginPortal()
        google()
    }
}
```
to my project build.gradle.  Honestly not sure why that worked, but it seems to be happy now

**Reply by jan-tennert** [reply]
Weird, but great that it works now!
